picard/util/checkupdate.py

```diff
@@ -0,0 +1,82 @@
+# -*- coding: utf-8 -*-
+#
+# Picard, the next-generation MusicBrainz tagger
+# Copyright (C) 2018 Bob Swift
+#
+# This program is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License
+# as published by the Free Software Foundation; either version 2
+# of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+
+from picard import (PICARD_VERSION_STR_SHORT, log)
+from picard.const import PICARD_URLS
+import picard.util.webbrowser2 as wb2
+from PyQt5.QtWidgets import QMessageBox
+#from PyQt5 import QtCore, QtGui, QtWidgets
+#from PyQt5.QtGui import QIcon
+import urllib.request
+import re
+
+_RE_TEST = r'.*infoText\s*=\s*\"v([0-9\.]*)'
+
+_NOTICE_TEXT = '''
+A new version of Picard is available.
+
+Your version: v%s
+New version: v%s
+
+Would you like to download the new version?
+'''
+
+latest_version = ""
+''' Use a module-level variable to store the latest release information
+    to avoid multiple calls to the web site in a single Picard session. '''
+
+
+def get_latest_version_number():
+    '''Scrapes the Picard home page to extract the latest release version number.'''
+    global latest_version
```

Comment:
Is a global mandatory ?

Reply:
I was using it to hold the value so that subsequent calls are not made to the web site.  I'm open to suggestions for an alternate way of doing this.  I don't particularly like using a global either.